guide/topics/discordjs/firstCommand.md

```diff
@@ -0,0 +1,163 @@
+`message` object has the following properties in it:
+- author - returns the [User](https://discord.js.org/#/docs/main/stable/class/User) that has sent the message.
+- channel - returns the [TextChannel](https://discord.js.org/#/docs/main/stable/class/TextChannel) that the message has been sent in.
+- content - returns the content of the message
+- guild - returns the [Guild](https://discord.js.org/#/docs/main/stable/class/Guild) (server) of where the message was sent in
+- id - returns the ID of the message - this is especially useful when saving a reference to the message
+
+And a few functions, too:
+- delete() - deletes the message
+- pin() - pins the message to the channel
+- react(emoji) - reacts to the message with an emoji object
+
+There are many more properties in the `message` class, however these are the most used ones. If you'd like to get to know each one of them, you can view it here: [Message Class on discord.js.org](https://discord.js.org/#/docs/main/stable/class/Message)
+
+Having these in mind, we can construct the following code:
+```javascript
+client.on("message", (message) => {
+  if(message.author.bot || !message.guild) return;
+  message.channel.send(`Hello, ${message.author.username}!`);
+});
+```
+
+To make sure that the bot will not fall in a loop responding to it's own messages, we have to add an `if` statement check, to check if the author is a bot. If so, `return` the code. The same goes for the second line, which check's for the guild of the message, as we don't want to listen to DM messages. Now, that we are sure that the bot will not fall into a loop and that guild is present on the object, we can move on.  In the first bit of the last line, we are looking at `message` and then accessing the `channel` property of it. That returns the channel that the message was sent in. Channels have a method `.send(content)` that sends a message to them - and it requires a parameter of what the content of that message should be.
+
+With this code, we can `send` a message in `channel` that the original `message` was sent. Inside the `.send()` function, we can find a template string - indicated with backticks as opposed to the normal quotes. This allowes us to embed some code in the message: `${message.author.username}`. Here, again, we are looking at `message`, but this time not the `channel`, but instead the `author` of that message. That returns a user that sent that message. Then, we are getting the username from it. Easy!
+
+With this, the bot should respond "Hello, [username]!", for example "Hello, NorteX!" if it sees any message on any channel.
+
+## Making a command
+In the first sentence of creating the event, I said that all commands are prefixed messages. So, let's try and make a very simple command handler.
+```javascript
+client.on("message", (message) => {
+  if(message.author.bot || !message.guild) return;
+  
+  const split = message.content.split(" ");
+  const command = split[0];
+  const args = split.slice(1);
```

Comment:
If you want to refactor this into one line, you could do this:
```js
const [command, ...args] = message.content.split(" ");
```

The example above declares variables from `message.content.split(" ")` via "destructuring assignment". Each variable represents an index in the array. So `const command = split(0);` is the same as the `command` variable destructured above.

For more information on destructuring assignment, check out the [`MDN Docs`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment).

Reply:
Just commit it yourself, if it hasn’t been added yet.